galette/templates/default/pages/transaction_form.html.twig

```diff
@@ -240,8 +240,8 @@
                 $('.contributions-selection .contribution_row input[type=checkbox]').hide();
 
                 //Initialize Fomantic components
-                $('.contributions-selection .dropdown').dropdown()
-                $('[id$="rangestart"], [id$="rangeend"]').calendar();
+                $('.contributions-selection .dropdown').dropdown();
+                {% include "elements/js/calendar.js.twig" with {selector: ".contributions-selection #start_date_filter, .contributions-selection #end_date_filter"} %}
```

Comment:
Don't change the default selector for the widget to work properly :

```suggestion
                {% include "elements/js/calendar.js.twig" %}
``` 

However, I noticed a strange behaviour (with or without the widget properly loaded) : when filtering by dates, the list in the modal is reloaded with the correct results, but when reopening the modal afterwards (with or without selecting a contribution), the results are different. Im my case, when first filtering in the modal, I get 2 results. And then, reopening the modal, only one result is displayed.

Reply:
I'll test this afternoon for the date picker, thanks.

Reply:
OK. From a transaction, I've opened contributions list popup and then filtered on a payment type. I can see here that several contributions are shown but should not:
- already linked to a transaction
- having an amount greater than remaining amount to dispatch in my transaction.

When I close the popup and then reopen it, the payment type filter is still applied; but other restrictions are correctly applied. This second is the correct result as far as I can see.

I also suspect an issue in session recording (like something stored in session but not used immediately); but that's kind of hard to be sure, I'll have to make extra tests.